### PR TITLE
Make GUI send last traceback on core timeout error

### PR DESCRIPTION
**Pull request by ichorid**
This should help with figuring out what causes timeout errors

### CODE REVIEW

**Comment by ichorid** [comment]
retest this please

**Comment by ichorid** [comment]
retest this please

**Comment by devos50** [comment]
@ichorid could you rebase this one on the latest commit?

**Comment by ichorid** [comment]
Well, using GitHub "rebase and merge" feature should make rebasing by hand unnecessary. Also, GitHub PR tries to merge the thing on top of the latest commit anyway. But we can retest it to be sure.

**Comment by ichorid** [comment]
retest this please

**Comment by devos50** [comment]
@ichorid The PR is still failing due to the [libtorrent error](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_win64/850/artifact/output/test_runners_output/5.err/*view*/). IIRC, our PR tester does not automatically apply these commits on top of the latest commit on the target branch, you have to do that yourself.

File: src/tribler-gui/tribler_gui/tribler_window.py [hunk 1]
```diff
@@ -123,6 +124,9 @@ def on_exception(self, *exc_info):
         if self.debug_window:
             self.debug_window.setHidden(True)
 
+        if exc_info[0] is CoreConnectTimeoutError:
+            exception_text = exception_text + self.core_manager.core_traceback
```
Comment (devos50) [inline src/tribler-gui/tribler_gui/tribler_window.py:128]: Nice solution! 👍

**Pull request merged**